server/middleware/auth.js

```diff
@@ -0,0 +1,24 @@
+const jwt = require('jsonwebtoken');
+const config = require('config');
+
+module.exports = function (req, res, next) {
+	const token = req.header('x-auth-token');
+
+	if (!token) {
+		return res
+			.status(401)
+			.json({ msg: 'No token, authorization denied' });
+	}
+
+	try {
+		const decoded = jwt.verify(
+			token,
+			config.get('jwtSecret')
+		);
+
+		req.user = decoded.user;
+		next();
+	} catch (err) {
```

Comment:
For the sake of code organisation it might be nice to leave it inside the try, even if, upon it causing an error, the error message will be inaccurate. It's a parameterless function that can actually be used to handle errors so I feel we're unlikely to have problems with this line.

Keen to put lines 5-11 inside the try.

Reply:
Have a look at this doc, [https://expressjs.com/en/guide/error-handling.html](https://expressjs.com/en/guide/error-handling.html).
We might not need any general try catches, as express would do that for us by default (potentially).